src/Model/Behavior/SearchBehavior.php

```diff
@@ -62,10 +62,18 @@ public function findSearch(Query $query, array $options)
                 'to be nested under key "search" in find() options.'
             );
         }
-
+        
         $filters = $this->_getAllFilters(Hash::get($options, 'collection', 'default'));
 
-        $params = $this->_flattenParams((array)$options['search']);
+        $flattenWhitelist = [];
+        foreach($filters as $filter) {
```

Comment:
Also the filters are keyed by filter name in `$fitlers` so you can avoid this extra loop all together. You can just use `$filters[$key]` (with not empty check) to get filter config inside the loop used in `_filterParams()`.